// @elevenlabs/elevenlabs-js/serialization/types/AstAndOperatorNodeChildrenItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AstAndOperatorNodeChildrenItem = void 0;
const serializers = __importStar(require("../index"));
const core = __importStar(require("../../core"));
const AstBooleanNode_1 = require("./AstBooleanNode");
const AstDynamicVariableNode_1 = require("./AstDynamicVariableNode");
const AstllmNode_1 = require("./AstllmNode");
const AstNumberNode_1 = require("./AstNumberNode");
const AstStringNode_1 = require("./AstStringNode");
exports.AstAndOperatorNodeChildrenItem = core.serialization
    .union("type", {
    and_operator: core.serialization.lazyObject(() => serializers.AstAndOperatorNode),
    boolean_literal: AstBooleanNode_1.AstBooleanNode,
    dynamic_variable: AstDynamicVariableNode_1.AstDynamicVariableNode,
    eq_operator: core.serialization.lazyObject(() => serializers.AstEqualsOperatorNode),
    gt_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOperatorNode),
    gte_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOrEqualsOperatorNode),
    llm: AstllmNode_1.AstllmNode,
    lt_operator: core.serialization.lazyObject(() => serializers.AstLessThanOperatorNode),
    lte_operator: core.serialization.lazyObject(() => serializers.AstLessThanOrEqualsOperatorNode),
    neq_operator: core.serialization.lazyObject(() => serializers.AstNotEqualsOperatorNode),
    number_literal: AstNumberNode_1.AstNumberNode,
    or_operator: core.serialization.lazyObject(() => serializers.AstOrOperatorNode),
    string_literal: AstStringNode_1.AstStringNode,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
